common/context.go

```diff
@@ -65,6 +66,23 @@ func GetProcessID(ctx context.Context) int {
 	return v // it will return zero on error
 }
 
+// WithRegisterCallback adds the function to register an event loop callback to ctx.
+func WithRegisterCallback(ctx context.Context, rcb RegisterCallbackFn) context.Context {
+	return context.WithValue(ctx, ctxKeyCallback, rcb)
+}
+
+// GetRegisterCallback returns the function to register an event loop callback from ctx.
+func GetRegisterCallback(ctx context.Context) RegisterCallbackFn {
+	v := ctx.Value(ctxKeyCallback)
+	if v == nil {
+		return nil
+	}
+	if rcb, ok := v.(RegisterCallbackFn); ok {
+		return rcb
+	}
+	return nil
+}
+
```

Comment:
We can simplify this with:

```suggestion
func GetRegisterCallback(ctx context.Context) RegisterCallbackFn {
	v := ctx.Value(ctxKeyCallback)
	if v, ok := v.(RegisterCallbackFn); ok {
		return v
	}
	return nil
}
```

Reply:
Resolved by e5076d5.